draft-ietf-dnsop-svcb-https.md

```diff
@@ -1228,12 +1228,18 @@ Section 12.1 of HSTS}}.
 
 ## Use of HTTPS RRs in other protocols
 
-All protocols employing "http://" or "https://" URLs SHOULD respect HTTPS RRs.
-For example, clients that
+All HTTP connections to named origins are eligible to use HTTPS RRs, even
+when HTTP is used as part of another protocol or without an explicit HTTP
+URL.  For example, clients that
 support HTTPS RRs and implement the altered WebSocket {{!WebSocket=RFC6455}}
 opening handshake from the W3C Fetch specification {{FETCH}} SHOULD use HTTPS RRs
 for the `requestURL`.
 
+When HTTP is used in a context where URLs or redirects are not applicable
+(e.g. connections to an HTTP proxy), clients that find a corresponding HTTPS RR
+SHOULD implement a security upgrade behavior equivalent to the one specified in
```

Comment:
I find this vague to the point of being confusing.  I've got RFC6797 open right now, and looking through it, I just don't know what this sentence is actually saying I should do.  That spec doesn't clearly describe what should be done in such cases.

Or is this one of those cases where the spec is being intentionally vague to let the clients do what they want to do as long as it's some reasonable behavior similar/same as what they do for HSTS? Maybe that's reasonable since this is a SHOULD.

Reply:
> I find this vague to the point of being confusing. I've got RFC6797 open right now, and looking through it, I just don't know what this sentence is actually saying I should do. That spec doesn't clearly describe what should be done in such cases.

This text renders as "... equivalent to the one specified in Section 9.5.", so it's not directly a reference to RFC 6797.  Does that help?

> Or is this one of those cases where the spec is being intentionally vague to let the clients do what they want to do as long as it's some reasonable behavior similar/same as what they do for HSTS? Maybe that's reasonable since this is a SHOULD.

Yes, it's deliberately vague, but if it's confusing that's still bad.

Reply:
> This text renders as "... equivalent to the one specified in Section 9.5.", so it's not directly a reference to RFC 6797. Does that help?

Ah, yes, that helps quite a bit.  I still think I would be a little bit confused on what clients should actually do if I were coming to this without all the context of our previous discussions since 9.5 mostly just talks about doing a redirect, but I don't have any ideas on how to do better without getting over-specified.  So I guess LGTM unless somebody has any better ideas.

Reply:
Perhaps we should just define it for the two cases in question? (I can only think of WebSockets and proxies.) Although it's messy because HTTP doesn't do a great job of defining proxies, and the IETF and WHATWG disagree on how to define WebSockets opening. (In WHATWG and browsers, it seems WebSockets cannot redirect. Poking around Chromium, looks like we do allow WebSockets to have HSTS and handle it somewhat specially here.)
https://fetch.spec.whatwg.org/#websocket-opening-handshake

Given all the other stuff going on in this draft, another option is to explicitly leave it undefined, but leave a little note that, if in an HTTP-like-but-not-quite-HTTP context where redirects aren't allowed or meaningful (e.g. WebSockets and proxies), that protocol should define an equivalent mechanism and use it. (Although, it's unclear whether this will ever get written down. For WebSockets, we probably need a rfc6455bis to realign it with reality. HTTP proxies probably needs a whole new document to define them better.)

Reply:
> that protocol should define an equivalent mechanism and use it

Will IETF let us get away with saying protocols should define something when relevant protocols already exist that don't define it?

Reply:
@davidben I think this text is close enough to that without getting too meta.